src/MainGrid.vala

```diff
@@ -76,30 +76,52 @@ public class Badger.MainGrid : Gtk.Grid {
         subheading.margin_bottom = 12;
         attach (subheading, 0, 1, 2, 1);
 
-        var labels = new Gtk.Label[reminders.length];
-        var scales = new Gtk.Scale[reminders.length];
+        HashTable<string, Scale> scales = new HashTable<string, Scale> (str_hash, str_equal);
+
+        // Change a single scale when the corresponding checkbox is pressed
+        settings.changed.connect ((key) => {
+            // just check for '-active' keys
+            if (key.has_suffix("-active")) {
+                bool value = settings.get_boolean(key);
+                scales.get(key).sensitive = value;
+            }
+        });
+
+        // Change all the scales when the global switch is pressed
+        global_switch.state_set.connect ((value) => {
+            global_switch.set_active(value);
+            scales.foreach ((key, scale) => {
+                scale.sensitive = value == true ? settings.get_boolean (key) : false;
```

Comment:
Since `value` is a boolean, we should essentially be able to say `value ?` instead of `value == true ?`.

Reply:
Yes you are right, that is a more concise way to write it.